pkg/virt-controller/watch/vm.go

```diff
@@ -3186,6 +3186,12 @@ func (c *VMController) sync(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachin
 			return vm, &syncErrorImpl{fmt.Errorf("Error encountered when trying to check if VMI has interface with ordinal names (e.g.: eth1, eth2..): %v", err), HotPlugNetworkInterfaceErrorReason}, nil
 		}
 		updatedVmiSpec := network.ApplyDynamicIfaceRequestOnVMI(vmCopy, vmiCopy, hasOrdinalIfaces)
+		// If a new interface is now present apply any device preferences we might have
+		if len(updatedVmiSpec.Domain.Devices.Interfaces) > len(vmiCopy.Spec.Domain.Devices.Interfaces) {
```

Comment:
I think this check is not sufficient, because it could be possible to add interfaces while removing others at the same time? If so the count total count of interfaces could be equal while they still have changed.

Just saw @EdDev's comment below. Are there ways to make this check more robust?

Reply:
Applying preferences is idempotent so we could just remove the conditional and always call `applyInferfacePreferences` in this path.